tuning/automation/ConvertToEfficiency.py

```diff
@@ -0,0 +1,116 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell cop-
+# ies of the Software, and to permit persons to whom the Software is furnished
+# to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+# PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+# CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+################################################################################
+
+import argparse
+import os
+import yaml
+
+typeIndexToName = {0: "S", 1: "D", 4: "H", 7: "B"}
+
+def parseArgs():
+    argParser = argparse.ArgumentParser()
+
+    h = {"inDir"   : "Directory containing input logic files", \
+         "outDir"  : "Output directory for modified logic files", \
+         "sclk"    : "SCLK frequency in MHz tuning was done at", \
+         "specs"   : ".yaml file containing hardware specifications", \
+         "per-cu"  : "If tuning was done per CU", \
+         "name"    : "Name substring to filter which files are modified", \
+         "mfma"    : "If MFMA instructions were used for tuning", \
+         "mi50"    : "For vega20, if tuning was done on mi50"
+    }
+
+    argParser.add_argument("inDir", metavar="input-dir", type=str, help=h["inDir"])
+    argParser.add_argument("outDir", metavar="output-dir", type=str, help=h["outDir"])
+    argParser.add_argument("sclk", type=int, help=h["sclk"])
+    argParser.add_argument("specs", metavar="hardware-specs", nargs="?", type=str, default="default_specs.yaml", help=h["specs"])
+    argParser.add_argument("-p", "--per-cu", action="store_true", help=h["per-cu"])
+    argParser.add_argument("-n", "--name", type=str, help=h["name"])
+    argParser.add_argument("-m", "--mfma", action="store_true", help=h["mfma"])
+    argParser.add_argument("--mi50", action="store_true", help=h["mi50"])
+
+    return argParser.parse_args()
+
+def allFiles(startDir):
+    current = os.listdir(startDir)
+    files = []
+    for filename in current:
+        fullPath = os.path.join(startDir, filename)
+        if os.path.isdir(fullPath):
+            files = files + allFiles(fullPath)
+        else:
+            files.append(fullPath)
+    return files
+
+def peakGFlops(sclk, alu, numCUs):
+    return (sclk / 1000) * alu * numCUs
+
```

Comment:
Might be a good idea just for numbers sake to indicate the units of sclk (=MHz in this case), alu (= wave size?)

Reply:
`alu` is Flops/cycle/CU here. I'll add a comment documenting the units.